akka-http-circe/src/main/scala/de/heikoseeberger/akkahttpcirce/CirceSupport.scala

```diff
@@ -17,50 +17,83 @@
 package de.heikoseeberger.akkahttpcirce
 
 import akka.http.scaladsl.marshalling.{ Marshaller, ToEntityMarshaller }
+import akka.http.scaladsl.model.HttpEntity
 import akka.http.scaladsl.model.MediaTypes.`application/json`
 import akka.http.scaladsl.unmarshalling.{ FromEntityUnmarshaller, Unmarshaller }
 import akka.util.ByteString
-import io.circe.{ jawn, Decoder, Encoder, Json, Printer }
+import io.circe.{ jawn, Decoder, Errors, Printer, RootEncoder }
 
 /**
   * Automatic to and from JSON marshalling/unmarshalling using an in-scope *Circe* protocol.
+  * The unmarshaller fails-fast, throwing the first `Error` encountered.
   *
-  * To use automatic codec derivation, user need to import `circe.generic.auto._`.
+  * To use automatic codec derivation, user need to import `io.circe.generic.auto._`.
   */
-object CirceSupport extends CirceSupport
+object CirceSupport extends CirceSupport with NoSpacesPrinter with FailFastUnmarshaller
```

Comment:
done, putting the annotation on the object, or would you prefer a `val` in the package object?